x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/warnings_popover.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import './workspace_panel_wrapper.scss';
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiPopover, EuiText, EuiButtonEmpty, EuiHorizontalRule } from '@elastic/eui';
+
+export const WarningsPopover = ({
+  children,
+}: {
+  children?: React.ReactNode | React.ReactNode[];
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  if (!children) {
+    return null;
+  }
+
+  const onButtonClick = () => setIsPopoverOpen((isOpen) => !isOpen);
+  const closePopover = () => setIsPopoverOpen(false);
+  const warningsCount = React.Children.count(children);
+  return (
+    <EuiPopover
+      button={
+        <EuiButtonEmpty
+          onClick={onButtonClick}
+          iconType="alert"
+          className="lnsWorkspaceWarningButton"
+        >
+          {i18n.translate('xpack.lens.chartWarnings.number', {
+            defaultMessage: `{warningsCount} {warningsCount, plural, one {warning} other {warnings}}`,
+            values: {
+              warningsCount,
+            },
+          })}
+        </EuiButtonEmpty>
+      }
+      isOpen={isPopoverOpen}
+      closePopover={closePopover}
+    >
+      {React.Children.map(children, (child, index) => (
+        <>
+          <EuiText style={{ width: 280 }}>{child}</EuiText>
+          {warningsCount - 1 !== index && <EuiHorizontalRule margin="s" />}
+        </>
+      ))}
```

Comment:
The contents of the `EuiPopover` don't current appear to have a maximum height applied where the popover contents will begin to allow vertical scrolling (for situations where a user has a large amount of warnings, for example). It currently just renders a very tall popover, which isn't ideal. Would it be possible to apply a maximum height and allow for vertical scrolling on a parent container?